### PR TITLE
added aggregating to queries

### PR DESCRIPTION
**Pull request by isiagi**
**What does the PR do**
The PR aggregate the queries
**What task does the PR solve**
optimization of the queries

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -310,7 +331,8 @@ class SmsView(ListCreateAPIView):
     def get_queryset(self):
         user = self.request.user
         if user.role == USER_TYPE_DHO:
-            model = SmsModel.objects.filter(sender_id=user.id).order_by('-created_at')
+            model = SmsModel.objects.filter(
```
Comment (codephillip) [inline app/views.py:334]: You should not be editing this file

File: app/views.py [hunk 2]
```diff
@@ -353,5 +375,6 @@ class AirtimeDispatchView(APIView):
 
     def post(self, request, format=None):
         airtime_module = AirtimeModule()
-        airtime_module.send_airtime(request.data["numbers"], request.data["amount"])
```
Comment (codephillip) [inline app/views.py:356]: You should not be editing this file

File: app/views.py [hunk 3]
```diff
@@ -50,11 +50,13 @@ def get_queryset(self):
...
-            model = Girl.objects.filter(Q(user__in=users) | Q(user=user)).order_by('-created_at')
+            model = Girl.objects.filter(Q(user__in=users) | Q(
+                user=user)).order_by('-created_at')
         elif user.role in [USER_TYPE_CHEW]:
             model = Girl.objects.filter(user=user).order_by('-created_at')
         elif user.role in [USER_TYPE_DHO]:
-            model = Girl.objects.filter(user__district=user.district).order_by('-created_at')
+            model = Girl.objects.filter(
+                user__district=user.district).order_by('-created_at')
         else:
```
Comment (codephillip) [inline app/views.py:60]: dont edit these lines

File: app/views.py [hunk 4]
```diff
@@ -69,11 +71,14 @@ def get_queryset(self):
...
         elif user.role in [USER_TYPE_CHEW]:
-            model = MappingEncounter.objects.filter(user=user).order_by('-created_at')
+            model = MappingEncounter.objects.filter(
+                user=user).order_by('-created_at')
         elif user.role in [USER_TYPE_DHO]:
-            model = MappingEncounter.objects.filter(user__district=user.district).order_by('-created_at')
+            model = MappingEncounter.objects.filter(
+                user__district=user.district).order_by('-created_at')
         else:
             model = MappingEncounter.objects.all().order_by('-created_at')
```
Comment (codephillip) [inline app/views.py:83]: dont edit these lines

File: app/views.py [hunk 5]
```diff
@@ -125,7 +130,8 @@ def get_queryset(self):
         user = self.request.user
 
         if user.role in [USER_TYPE_DHO]:
-            model = HealthFacility.objects.filter(sub_county__county__district=user.district)
+            model = HealthFacility.objects.filter(
+                sub_county__county__district=user.district)
         else:
```
Comment (codephillip) [inline app/views.py:135]: dont edit these lines

File: app/views.py [hunk 6]
```diff
@@ -139,11 +145,14 @@ def get_queryset(self):
...
+                Q(user__in=users) | Q(user=user)).order_by('-created_at')
         elif user.role == USER_TYPE_CHEW:
-            model = FollowUp.objects.filter(Q(user=user)).order_by('-created_at')
+            model = FollowUp.objects.filter(
+                Q(user=user)).order_by('-created_at')
         elif user.role == USER_TYPE_DHO:
-            model = FollowUp.objects.filter(user__district=user.district).order_by('-created_at')
+            model = FollowUp.objects.filter(
+                user__district=user.district).order_by('-created_at')
         else:
```
Comment (codephillip) [inline app/views.py:156]: dont edit these lines

File: app/views.py [hunk 7]
```diff
@@ -168,7 +177,8 @@ def get_queryset(self):
         elif user.role == USER_TYPE_CHEW:
             model = Delivery.objects.filter(user=user).order_by('-created_at')
         elif user.role == USER_TYPE_DHO:
-            model = Delivery.objects.filter(user__district=user.district).order_by('-created_at')
+            model = Delivery.objects.filter(
+                user__district=user.district).order_by('-created_at')
         else:
```
Comment (codephillip) [inline app/views.py:182]: dont edit these lines

File: app/views.py [hunk 8]
```diff
@@ -194,10 +204,12 @@ def get_queryset(self):
...
             # return appointments created by CHEW
-            appointments = Appointment.objects.filter(Q(user=user) | Q(girl__user=user)).order_by('-created_at')
+            appointments = Appointment.objects.filter(
+                Q(user=user) | Q(girl__user=user)).order_by('-created_at')
         elif user.role in [USER_TYPE_DHO]:
             # return all appointments in the DHO district
-            appointments = Appointment.objects.filter(user__district=user.district).order_by('-created_at')
+            appointments = Appointment.objects.filter(
+                user__district=user.district).order_by('-created_at')
         else:
```
Comment (codephillip) [inline app/views.py:213]: dont edit these lines

**Review by codephillip — COMMENTED** [review]
Please only edit what is needed. You are editing files which are not part of the commit

File: app/views.py (hunk 1 above)
Comment (isiagi) [reply app/views.py:334]: I think when I formatted the document. some work disappeared.  i never met to do that

**Pull request closed without merging**